HiP-DataStore/Core/WriteModel/ScoreBoardIndex.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using PaderbornUniversity.SILab.Hip.DataStore.Model.Events;
+using PaderbornUniversity.SILab.Hip.DataStore.Model.Entity;
+
+namespace PaderbornUniversity.SILab.Hip.DataStore.Core.WriteModel
+{
+    public class ScoreBoardIndex : IDomainIndex
+    {
+        private readonly ScoreBoard _board = new ScoreBoard();
+        private readonly Object _lockObject = new object();
+
+        private int _maximumId;
+        public int NewId()
+        {
+            return ++_maximumId;
+        }
+
+        //All records of ScoreBoard (Sorted)
+        public IReadOnlyCollection<ScoreRecord> AllRecords()
+        {
+            return _board.ToList();
+        }
+        public bool Exists(int userId)
+        {
+            return _board.Any(x => x.UserId == userId);
+        }
+
+
+        public void ApplyEvent(IEvent e)
+        {
+            switch (e)
+            {
+                case ScoreAdded ev:
+                    lock (_lockObject)
```

Comment:
This `lock`-statement is useless, unless you apply it everywhere where `_board` or `_maximumId` are accessed. For example it could still happen (in theory) that you are concurrently reading (in line 27) and modifying (line 39/40) `_boards`, because the `lock`-statement is missing around line 27.

Practically though, I have not yet experienced this issue with our DataStore, but you never know ;)